src/app_service/service/community/service.nim

```diff
@@ -1528,3 +1604,53 @@ QtObject:
     except Exception as e:
       error "Error extracting discord channels and categories", msg = e.msg
 
+  proc editCommunityTokenPermission*(self: Service, communityId: string, tokenPermission: CommunityTokenPermissionDto) =
+    try:
+      let editing = tokenPermission.id != ""
+
+      var response: RpcResponse[JsonNode]
+
+      if editing:
+        response = status_go.editCommunityTokenPermission(communityId, tokenPermission.id, int(tokenPermission.`type`), Json.encode(tokenPermission.tokenCriteria), tokenPermission.isPrivate)
+      else:
+        response = status_go.createCommunityTokenPermission(communityId, int(tokenPermission.`type`), Json.encode(tokenPermission.tokenCriteria), tokenPermission.isPrivate)
+
+      if response.result != nil and response.result.kind != JNull:
+        var changesField = "tokenPermissionsAdded"
+        if editing:
+          changesField = "tokenPermissionsModified"
+
+        for permissionId, permission in response.result["communityChanges"].getElems()[0][changesField].pairs():
+          let p = permission.toCommunityTokenPermissionDto()
+          self.allCommunities[communityId].tokenPermissions[permissionId] = p
+          self.joinedCommunities[communityId].tokenPermissions[permissionId] = p
```

Comment:
Ah you actually already do the thing I commented above. The check for `community.admin` is then double useless :smile: